pkg/cloud/services/loadbalancer/client.go

```diff
@@ -0,0 +1,231 @@
+	CreateListener(opts listeners.CreateOptsBuilder) (*listeners.Listener, error)
+	ListListeners(opts listeners.ListOptsBuilder) ([]listeners.Listener, error)
+	GetListener(id string) (*listeners.Listener, error)
+	DeleteListener(id string) error
+	CreatePool(opts pools.CreateOptsBuilder) (*pools.Pool, error)
+	ListPools(opts pools.ListOptsBuilder) ([]pools.Pool, error)
+	GetPool(id string) (*pools.Pool, error)
+	DeletePool(id string) error
+	CreatePoolMember(poolID string, opts pools.CreateMemberOptsBuilder) (*pools.Member, error)
+	ListPoolMember(poolID string, opts pools.ListMembersOptsBuilder) ([]pools.Member, error)
+	DeletePoolMember(poolID string, lbMemberID string) error
+	CreateMonitor(opts monitors.CreateOptsBuilder) (*monitors.Monitor, error)
+	ListMonitors(opts monitors.ListOptsBuilder) ([]monitors.Monitor, error)
+	DeleteMonitor(id string) error
+	ListLoadBalancerProviders() ([]providers.Provider, error)
+}
+
+type lbClient struct {
+	serviceClient *gophercloud.ServiceClient
+}
+
+func (l lbClient) CreateLoadBalancer(opts loadbalancers.CreateOptsBuilder) (*loadbalancers.LoadBalancer, error) {
+	mc := metrics.NewMetricPrometheusContext("loadbalancer", "create")
+	lb, err := loadbalancers.Create(l.serviceClient, opts).Extract()
+	if mc.ObserveRequest(err) != nil {
+		return nil, err
+	}
+	return lb, nil
+}
+
+func (l lbClient) ListLoadBalancers(opts loadbalancers.ListOptsBuilder) ([]loadbalancers.LoadBalancer, error) {
+	mc := metrics.NewMetricPrometheusContext("loadbalancer", "list")
+	allPages, err := loadbalancers.List(l.serviceClient, opts).AllPages()
+	if mc.ObserveRequest(err) != nil {
+		return nil, err
+	}
+	return loadbalancers.ExtractLoadBalancers(allPages)
+}
+
+func (l lbClient) GetLoadBalancer(id string) (*loadbalancers.LoadBalancer, error) {
+	mc := metrics.NewMetricPrometheusContext("loadbalancer", "get")
+	lb, err := loadbalancers.Get(l.serviceClient, id).Extract()
+	if mc.ObserveRequest(err) != nil {
+		return nil, err
+	}
+	return lb, nil
+}
+
+func (l lbClient) DeleteLoadBalancer(id string, opts loadbalancers.DeleteOptsBuilder) error {
+	mc := metrics.NewMetricPrometheusContext("loadbalancer", "delete")
+	err := loadbalancers.Delete(l.serviceClient, id, opts).ExtractErr()
+	if mc.ObserveRequestIgnoreNotFound(err) != nil && !capoerrors.IsNotFound(err) {
+		return err
+	}
+	return nil
+}
+
+func (l lbClient) CreateListener(opts listeners.CreateOptsBuilder) (*listeners.Listener, error) {
+	mc := metrics.NewMetricPrometheusContext("loadbalancer_listener", "create")
+	listener, err := listeners.Create(l.serviceClient, opts).Extract()
+	if mc.ObserveRequest(err) != nil {
+		return nil, err
+	}
+	return listener, nil
+}
+
+func (l lbClient) ListListeners(opts listeners.ListOptsBuilder) ([]listeners.Listener, error) {
+	mc := metrics.NewMetricPrometheusContext("loadbalancer_listener", "list")
+	allPages, err := listeners.List(l.serviceClient, opts).AllPages()
```

Comment:
There are some occurences of `allPages` as variable name. I'd suggest to rename those to align more with the items that are in the list, e.g.
```suggestion
	listenerList, err := listeners.List(l.serviceClient, opts).AllPages()
```

Reply:
Sorry, the return value is not a list but rather pages. And this seems to be used like this throughout the repository. So it's perfectly fine like this.